lib/archethic_web/live/chains/beacon_live.ex

```diff
@@ -79,33 +81,32 @@ defmodule ArchethicWeb.BeaconChainLive do
     page = Map.get(params, "page", "1")
 
     case Integer.parse(page) do
-      {number, ""} when number > 0 and is_list(dates) ->
-        if number > length(dates) do
-          {:noreply,
-           push_redirect(socket, to: Routes.live_path(socket, __MODULE__, %{"page" => 1}))}
-        else
-          new_assign =
-            socket
-            |> assign(:current_date_page, number)
-            |> assign(:transactions, [])
-            |> assign(:fetching, true)
-
-          if Enum.at(dates, 0) == Enum.at(dates, number - 1) do
-            next_summary_date =
-              dates
-              |> Enum.at(number - 1)
-
-            send(self(), {:initial_load, next_summary_date})
-          else
-            date =
-              dates
-              |> Enum.at(number - 1)
-
-            send(self(), {:load_at, date})
-          end
-
-          {:noreply, new_assign}
-        end
+      {1, ""} ->
+        send(self(), :initial_load)
+
+        new_assign =
+          socket
+          |> assign(:current_date_page, 1)
+          |> assign(:transactions, [])
+          |> assign(:fetching, true)
+
+        {:noreply, new_assign}
+
+      {number, ""} when number > length(dates) ->
+        {:noreply,
+         push_redirect(socket, to: Routes.live_path(socket, __MODULE__, %{"page" => 1}))}
+
+      {number, ""} ->
+        new_assign =
+          socket
+          |> assign(:current_date_page, number)
+          |> assign(:transactions, [])
+          |> assign(:fetching, true)
+
+        date = Enum.at(number, -1)
```

Comment:
Here ,
`date = Enum.at(dates, number -1)`

Reply:
Oh yes. Problem of rebase thanks